File: elixir/all_your_base.ex
defmodule AllYourBase do
  @doc """
  Given a number in input base, represented as a sequence of digits, converts it to output base,
  or returns an error tuple if either of the bases are less than 2
  """
  @spec convert(list, integer, integer) :: {:ok, list} | {:error, String.t()}
  def convert(digits, input_base, output_base) do
    cond do
      output_base < 2 ->
        {:error, "output base must be >= 2"}

      input_base < 2 ->
        {:error, "input base must be >= 2"}

      Enum.any?(digits, &(&1 < 0 || &1 >= input_base)) ->
        {:error, "all digits must be >= 0 and < input base"}

      true ->
        digits
        |> Enum.reduce(0, fn d, acc ->
          acc * input_base + d
        end)
        |> aux(output_base, [])
    end
  end

  defp aux(0, _, []), do: {:ok, [0]}
  defp aux(0, _, lst), do: {:ok, lst}

  defp aux(n, base, lst) do
    aux(div(n, base), base, [rem(n, base) | lst])
  end
end

File: python/all_your_base.py
from functools import reduce


def rebase(input_base, digits, output_base):
    if input_base < 2:
        raise ValueError("input base must be >= 2")

    if output_base < 2:
        raise ValueError("output base must be >= 2")

    if any([True for d in digits if d < 0 or d >= input_base]):
        raise ValueError("all digits must satisfy 0 <= d < input base")

    n = reduce(lambda x, y: x * input_base + y, digits, 0)

    if n == 0:
        return [0]
    
    output_digits = []
    while n != 0:
        output_digits.append(n % output_base)
        n //= output_base

    return output_digits[::-1]
    
